internal/web/cacheverifier.go

```diff
@@ -0,0 +1,250 @@
+			panic(err)
+		}
+		verifier.setClients(cachedClient, nonCachedClient)
+
+		tick := time.NewTicker(time.Second * time.Duration(8*(rand.Float32()+1))) // between 8 - 16 seconds
+		tickC := tick.C
+		defer tick.Stop()
+
+		var diffErr error
+
+	endlessLoop:
+		for range tickC {
+			if nonCachedItems, err := verifier.listNonCachedItems(ctx); err != nil {
+				fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed to get actual items: %v\n", err)
+			} else if cachedItems, err := verifier.listCachedItems(ctx); err != nil {
+				fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed get cached items: %v\n", err)
+			} else {
+			outerLoop:
+				for _, nonCachedItem := range nonCachedItems {
+					for _, cachedItem := range cachedItems {
+						if nonCachedItem.GetNamespace() == cachedItem.GetNamespace() &&
+							nonCachedItem.GetName() == cachedItem.GetName() {
+							if nonCachedItem.GetName() == "glasskube" {
+								nonCachedItem.SetResourceVersion("asdf")
+							}
+							if nonCachedItem.GetResourceVersion() != cachedItem.GetResourceVersion() {
+								fmt.Fprintf(os.Stderr, "CACHEVERIFIER: resource versions of %s differ, need to check\n",
+									cachedItem.GetName())
+								if diffErr = verifier.handleDiff(cachedItem, nonCachedItem); diffErr != nil {
+									break endlessLoop
+								}
+							}
+							continue outerLoop
+						}
+					}
+					fmt.Fprintf(os.Stderr, "CACHEVERIFIER: non-cached item %s not found in cache, need to check\n",
+						nonCachedItem.GetName())
+					// non cached package found, that is not in cache (yet?)
+					if diffErr = verifier.handleDiff(nil, nonCachedItem); diffErr != nil {
+						break endlessLoop
+					}
+				}
+
+				// check the other way around (i.e. whether there are cached items, that are not in the non-cached list anymore)
+				for _, cachedClPkg := range cachedItems {
+					found := false
+				innerLoop:
+					for _, nonCachedClPkg := range nonCachedItems {
+						if nonCachedClPkg.GetNamespace() == cachedClPkg.GetNamespace() &&
+							nonCachedClPkg.GetName() == cachedClPkg.GetName() {
+							found = true
+							break innerLoop
+						}
+					}
+					if !found {
+						fmt.Fprintf(os.Stderr, "CACHEVERIFIER: cached item %s not found in non-cached list, need to check\n",
+							cachedClPkg.GetName())
+						// when here, there exists a cached item, that is not in the non-cached list
+						if diffErr = verifier.handleDiff(cachedClPkg, nil); diffErr != nil {
+							break endlessLoop
+						}
+					}
+				}
+			}
+
+		}
+		verifier.closeAndReset(diffErr)
+	}()
+}
+
+func (verifier *cacheVerifier) closeAndReset(err error) {
+	verifier.mutex.Lock()
+	defer verifier.mutex.Unlock()
+	close(verifier.stopCh)
+	clear(verifier.diffMap)
+	verifier.errCh <- err
+}
+
+func (verifier *cacheVerifier) newStopCh() chan struct{} {
+	verifier.stopCh = make(chan struct{})
+	return verifier.stopCh
+}
+
+func (verifier *cacheVerifier) handleDiff(cachedObj metav1.Object, nonCachedObj metav1.Object) error {
```

Comment:
True, any error would have been considered as a failed verification, while it could be an "actual" error. I changed it to a custom error type now, and check for that. 
The idea of using an error was to transport detailed information to the outside (e.g. logs and telemetry), and from my perspective that should be reasonable: We do something similar when we have http errors for example, we also check for the status code there and make our subsequent logic dependent on it. Same for our configuration validations. But maybe I am missing something?